include/storage/flash_map.h

```diff
@@ -111,6 +111,9 @@ int flash_area_check_int_sha256(const struct flash_area *fa,
  * @param[in]  id ID of the flash partition.
  * @param[out] fa Pointer which has to reference flash_area. If
  * @p ID is unknown, it will be NULL on output.
+ *
+ * @return  0 on success, -EACCES if the flash_map is not available ,
+ * -ENOENT if @p ID is unknown.
```

Comment:
shouldn't you use here "id" instead of ID? 
Also, since you're here, maybe change "p" to "param" above